pageserver/src/http/routes.rs

```diff
@@ -355,112 +355,100 @@ async fn get_lsn_by_timestamp_handler(request: Request<Body>) -> Result<Response
 async fn tenant_attach_handler(request: Request<Body>) -> Result<Response<Body>, ApiError> {
     let tenant_id: TenantId = parse_request_param(&request, "tenant_id")?;
     check_permission(&request, Some(tenant_id))?;
+    let allow_local_files = query_param_present(&request, "allow-local-files");
 
-    info!("Handling tenant attach {tenant_id}");
+    info!("Handling tenant attach {tenant_id}, local files allowed: {allow_local_files}");
 
-    tokio::task::spawn_blocking(move || match tenant_mgr::get_tenant(tenant_id, false) {
+    let mut attaching_empty_tenant = false;
```

Comment:
I start to dislike this special case more and more; looks like the only "valid" timeline-less state of a tenant is when we create one (it gets created without timelines).
Otherwise, console operations never allow us to have a tenant without timelines, so should we remove such tenants on startup and never allow in the pageserver?

Might bring the "initial timeline" concept back, we used to create both tenant and a timeline before removed this.